src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9251,6 +9248,9 @@ private ExpressionSyntax ParseTerm(Precedence precedence)
                 case SyntaxKind.NewKeyword:
                     expr = this.ParseNewExpression();
                     break;
+                case SyntaxKind.StackAllocKeyword:
+                    expr = this.ParseStackAllocExpression();
+                    break;
```

Comment:
I don't understand what precedence this expression is at in the syntax hierarchy. In an expression such as `stackalloc int[10][2]`, is that indexing the expression `stackalloc int[10]` with the index `2`? I think we need a proposed language grammar that makes it clear what is going on here, which we can review the parser changes against.